### ISSUE TITLE
Show warning only once

### ISSUE THREAD

**Issue by slavafomin** [sass/sass#2568]
Hello!

I have a library with SASS file in which I want to show the warning message, but I want to do it only once regardless of how many times this file is imported.

If I just add the `@warn` directive, it will be displayed for each import of the source file.

I've tried to wrap it in `@if` statement and to use the flag to control this behavior, but I can't make it work for some reason:

```scss

@if (global-variable-exists(app-file-imported) == false) {
  @warn 'SOME WARNING';
}

// some code ...

$app-file-imported: true;

```

I've tried both `variable-exists` and `global-variable-exists` functions, but the warning statement always executes.

It would be very nice to have a construct like `@warn-once`, which will raise the warning only once, regardless of how many times the source file is imported.

**Comment by nex3** [comment]
This will be taken care of with the import-once behavior of the new module system (#139). For now, you can do this:

```scss
$-app-file-imported: false !default;
@if not $-app-file-imported {
  @warn "Some warning";
}
$-app-file-imported: true;
```

or even:

```scss
$-warnings-emitted: ();

@mixin warn-once($warning) {
  @if not map-get($-warnings-emitted, $warning) {
    @warn $warning;
    $-warnings-emitted: map-set($warnings-emitted, $warning, true);
  }
}

**Comment by slavafomin** [comment]
@nex3 thank you for the examples. However, they doesn't seem to work in my environment. The warnings are displayed for each import (both examples). It looks like variables are reinitialized on every call. Could this be caused by the fact, that I'm using LibSass (node-sass)?

BTW, in the second example you have a typo: it should be `$-warnings-emitted` instead of `$warnings-emitted`.

Also, I have a feeling that first line: `$-warnings-emitted: ()` would reset the map to an empty one on each import. Isn't it?

And I can't find any documentation for `map-set()` function.

**Comment by nex3** [comment]
Sorry, I didn't test that example. Here's a version that works:

```scss
$-warnings-emitted: () !default;

@mixin warn-once($warning) {
  @if not map-get($-warnings-emitted, $warning) {
    @warn $warning;
    $-warnings-emitted: map-merge($-warnings-emitted, ($warning: true)) !global;
  }
}

@include warn-once("Some warning");
```

**Comment by slavafomin** [comment]
Thanks! But, for some reason I continue to see the duplicate warnings, even with your latest example...
